paseos/visualization/space_animation.py

```diff
@@ -0,0 +1,346 @@
+                            obj.positions[-1, :] = pos_norm
+                        else:
+                            obj.positions = np.vstack((obj.positions, pos_norm))
+                    else:
+                        obj.positions = np.array(pos_norm)
+        self._plot_actors()
+
+        # Step through trajectories to find max and min values in each direction
+        coords_max = [
+            self.ax_3d.get_xlim()[1],
+            self.ax_3d.get_ylim()[1],
+            self.ax_3d.get_zlim()[1],
+        ]
+        coords_min = [
+            self.ax_3d.get_xlim()[0],
+            self.ax_3d.get_ylim()[0],
+            self.ax_3d.get_zlim()[0],
+        ]
+        for obj in self.objects:
+            coords_max = np.maximum(obj.positions.max(axis=0), coords_max)
+            coords_min = np.minimum(obj.positions.min(axis=0), coords_min)
+        self.ax_3d.set_xlim(coords_min[0], coords_max[0])
+        self.ax_3d.set_ylim(coords_min[1], coords_max[1])
+        self.ax_3d.set_zlim(coords_min[2], coords_max[2])
+
+        # Update LOS heatmap
+        current_actors = list(current_actors)
+        los_matrix = self._get_los_matrix(current_actors)
+        self._los_plot.set_data(los_matrix)
+        xaxis = np.arange(len(current_actors))
+        self.ax_los.set_xticks(xaxis)
+        self.ax_los.set_yticks(xaxis)
+        self.ax_los.set_xticklabels(current_actors)
+        self.ax_los.set_yticklabels(current_actors)
+        self.fig.canvas.draw_idle()
+
+        self.ax_3d.set_title(self._sec_to_ddhhmmss(local_time_d / pk.SEC2DAY))
+
+    def _animate(self, sim: PASEOS, dt: float) -> List[Artist]:
+        """Advances the time of sim, updates the plot, and returns the axis objects
+
+        Args:
+            sim (PASEOS): simulation object.
+            dt (float): size of time step
+
+        Returns:
+            List[Artist]: list of Artist objects
+        """
+        sim.advance_time(dt)
+        self._update(sim)
+        return self.ax_3d.get_children() + self.ax_los.get_children()
+
+    def _animation_wrapper(self, step: int, sim: PASEOS, dt: float) -> List[Artist]:
+        """Wrapper to allow for frame numbers from animation
+
+        Args:
+            step (int): current frame number of the animation
+            sim (PASEOS): simulation object.
+            dt (float): size of time step
+
+        Returns:
+            List[Artist]: list of Artist objects
+        """
+        return self._animate(sim, dt)
+
+    def animate(self, sim: PASEOS, dt: float, steps: int = 1, name: str = None) -> None:
+        """Animates paseos for a given number of steps with dt in each step.
+
+        Args:
+            sim (PASEOS): simulation object.
+            dt (float): size of time step
+            steps (int, optional): number of steps to animate. Defaults to 1.
+            name (str, optional): filename to save the animation. Defaults to None.
```

Comment:
can we rename `name` -> `save_to_file` ? Otherwise could be mistaken for windowtitle